common/src/main/java/com/wynntils/screens/WynntilsQuestBookScreen.java

```diff
@@ -0,0 +1,513 @@
+                    .withStyle(ChatFormatting.GRAY));
+
+    private Widget hovered = null;
+    private final QuestBookSearchWidget searchWidget;
+    private int currentPage = 0;
+    private int maxPage = 0;
+    private List<QuestInfo> quests;
+    private List<QuestButton> questButtons = new ArrayList<>();
+    private QuestInfo tracked = null;
+
+    public WynntilsQuestBookScreen() {
+        super(new TranslatableComponent("screens.wynntils.wynntilsQuestBook.name"));
+
+        this.searchWidget = new QuestBookSearchWidget(
+                (int) (Texture.QUEST_BOOK_BACKGROUND.width() / 2f + 15),
+                0,
+                Texture.QUEST_BOOK_SEARCH.width(),
+                Texture.QUEST_BOOK_SEARCH.height(),
+                this::updateQuestsFilter,
+                this);
+
+        WynntilsMod.getEventBus().register(this);
+    }
+
+    @Override
+    public void onClose() {
+        WynntilsMod.getEventBus().unregister(this);
+        super.onClose();
+    }
+
+    @Override
+    protected void init() {
+        QuestBookManager.rescanQuestBook();
+        this.setQuests(QuestBookManager.getQuests());
+
+        this.addRenderableWidget(searchWidget);
+
+        this.addRenderableWidget(new BackButton(
+                (int) ((Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 16) / 2f),
+                65,
+                Texture.BACK_ARROW.width() / 2,
+                Texture.BACK_ARROW.height(),
+                new WynntilsMenuScreen()));
+
+        this.addRenderableWidget(new ReloadQuestsButton(
+                Texture.QUEST_BOOK_BACKGROUND.width() - 21,
+                11,
+                (int) (Texture.RELOAD_BUTTON.width() / 2 / 1.7f),
+                (int) (Texture.RELOAD_BUTTON.height() / 1.7f),
+                this));
+        this.addRenderableWidget(new PageSelectorButton(
+                Texture.QUEST_BOOK_BACKGROUND.width() / 2 + 50 - Texture.FORWARD_ARROW.width() / 2,
+                Texture.QUEST_BOOK_BACKGROUND.height() - 25,
+                Texture.FORWARD_ARROW.width() / 2,
+                Texture.FORWARD_ARROW.height(),
+                false,
+                this));
+        this.addRenderableWidget(new PageSelectorButton(
+                Texture.QUEST_BOOK_BACKGROUND.width() - 50,
+                Texture.QUEST_BOOK_BACKGROUND.height() - 25,
+                Texture.FORWARD_ARROW.width() / 2,
+                Texture.FORWARD_ARROW.height(),
+                true,
+                this));
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        renderBackgroundTexture(poseStack);
+
+        // Make 0, 0 the top left corner of the rendered quest book background
+        poseStack.pushPose();
+        final float translationX = getTranslationX();
+        final float translationY = getTranslationY();
+        poseStack.translate(translationX, translationY, 1f);
+
+        renderTitle(poseStack);
```

Comment:
There seem to be a fair amount of overlap between this class and `WynntilsMenuScreen`. Do you think they could share a common abstract ancestor that takes care of the "questbooky" things of the screen?